rpc/internal/test_results_service.go

```diff
@@ -41,18 +43,73 @@ func (s *testResultsService) CreateTestResultsRecord(ctx context.Context, info *
 
 // AddTestResults adds test results to an existing test results record.
 func (s *testResultsService) AddTestResults(ctx context.Context, results *TestResults) (*TestResultsResponse, error) {
-	return nil, newRPCError(codes.Unimplemented, errors.New("not implemented"))
+	record := &model.TestResults{ID: results.TestResultsRecordId}
+	record.Setup(s.env)
+	if err := record.Find(ctx); err != nil {
+		if db.ResultsNotFound(err) {
+			return nil, newRPCError(codes.NotFound, err)
+		}
+		return nil, newRPCError(codes.Internal, errors.Wrapf(err, "problem finding test results record for '%s'", results.TestResultsRecordId))
+	}
+
+	exportedResults := make([]model.TestResult, len(results.Results))
+	for i, result := range results.Results {
+		exportedResult, err := result.Export()
+		if err != nil {
+			return nil, newRPCError(codes.InvalidArgument, errors.Wrapf(err, "problem exporting test result"))
+		}
+		exportedResults[i] = exportedResult
+	}
+
+	return &TestResultsResponse{TestResultsRecordId: record.ID},
+		newRPCError(codes.Internal, errors.Wrapf(record.Append(ctx, exportedResults), "problem appending test results for '%s'", results.TestResultsRecordId))
 }
 
 // StreamTestResults adds test results via client-side streaming to an existing
 // test results record.
 func (s *testResultsService) StreamTestResults(stream CedarTestResults_StreamTestResultsServer) error {
-	return newRPCError(codes.Unimplemented, errors.New("not implemented"))
+	ctx := stream.Context()
+	id := ""
+
+	for {
+		if err := ctx.Err(); err != nil {
+			return newRPCError(codes.Aborted, err)
+		}
+
+		results, err := stream.Recv()
+		if err == io.EOF {
+			return stream.SendAndClose(&TestResultsResponse{TestResultsRecordId: id})
+		}
+		if err != nil {
+			return err
```

Comment:
should we wrap?

Reply:
the stream.Recv() func returns an rpc error code, should we still wrap it?